### ISSUE TITLE
Some backups don't contain all repositories

### ISSUE THREAD

**Issue by Gaibhne** [sameersbn/docker-gitlab#1364]
Title says it all, some of our backups are missing some of our repositories. I have not been able to discern a pattern in which repos are included and which ones aren't, but it's always the same ones missing.

Backups created with `docker exec --user git -it gitlab_gitlab_1 bundle exec rake gitlab:backup:create` are always complete, only the automated backups show this behaviour.

Our automated backup settings:

```
        - GITLAB_BACKUP_SCHEDULE=daily
        - GITLAB_BACKUP_TIME=23:00
        - GITLAB_BACKUP_EXPIRY=0
```

We are running `sameersbn/gitlab:9.5.5`.

Please advise what further info we can provide and/or how we can debug this issue ourselves, as we could find no log files related to the backup process.

**Comment by Biegal** [comment]
I've just found out I have the exact same issue.
I use GitLab for personal use, so I didn't found the issue earlier. What I can see from my backups is that it seems it worked fine when I had vers 9.1.2
Then I've upgraded to 9.3.5 and two things changed:
1) name of the backup file (I can see it has a version number in the filename now)
2) backups become much smaller and doesn't contain all repositories.
I've just upgraded to latest 10.0.2 version, I will check what backups will contain in following days.

My config:
    - GITLAB_BACKUPS=daily
    - GITLAB_BACKUP_DIR=/home/git/backups
    - GITLAB_BACKUP_EXPIRY=1814400

**Comment by Gaibhne** [comment]
This is still happening to me with 10.0.3.

When switching to running backups manually via the Docker hosts cron, I ran into an issue that once googled suggested that there is a problem with the backup folder being a Windows Samba share. I don't know if it that is related; @Biegal are you running your backups to a Windows share by any chance ?

**Comment by Biegal** [comment]
@Gaibhne no, I don't even have samba on that machine as far as I know. I'm using rsync to get the backups out of it.

**Comment by solidnerd** [comment]
Hey @Gaibhne  and @Biegal, did you see #1190 ?

I think this could fix your problem in future.

**Comment by Gaibhne** [comment]
I'm not sure this is what caused our problem. We have a server script that automatically docker-compose's down & up our GitLab during the night, and usually, we don't restart it other than that, so if I understand correctly, we should not have had duplicate crontab entries.

Also, since #1190 looks like it would cause race conditions, it seems to me like that should have caused our backups to be erratic, but instead I often would end up with many days in a row missing the exact same repos every day.

Nevertheless, if that fix ever gets merged in, I'll be happy to try if it fixes this issue.
